packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkStateManager.ts

```diff
@@ -468,4 +428,19 @@ export class ForkStateManager implements EIP2929StateManager {
     this._stateRoot = newRoot;
     this._state = state;
   }
+
+  public async hasStateRoot(root: Buffer): Promise<boolean> {
+    return this._state.has(bufferToHex(root));
+  }
+
+  public async flush(): Promise<void> {
+    // not implemented
+  }
+
+  public async modifyAccountFields(
+    _address: Address,
+    _accountFields: any
+  ): Promise<void> {
+    // not implemented
```

Comment:
We should probably throw

Reply:
I still think that this one should throw and kill hardhat network if called.  This nop here would lead to a bug if someone uses it, wouldn't it?

Reply:
Actually, it's called here: https://github.com/ethereumjs/ethereumjs-monorepo/blob/8cc931c031798b7ccb5c927286ea17043db907b5/packages/evm/src/interpreter.ts#L966

We need to implement it.

Reply:
Fixed in 917ba24e9. Added tests for this in ac5be4b4e.